### PR TITLE
Story/materialize client permissions

### PR DESCRIPTION
**Pull request by amichal**

### CODE REVIEW

**Comment by amichal** [comment]
@eanders This is at a place where i'd like to run it on a real-ish sample of users. My local db has only 11k clients and 1 user with full admin. As a result the job above runs in less than 2 seconds. A more realistic mix of users would be good.

File: app/jobs/materialize_permissions.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+require 'pp'
+require 'benchmark'
+
+class MaterializePermissions < BaseJob
+  COLS = ['user_id', 'client_id', 'viewable'].freeze
+
+  def perform
+    # WIP:
+    # TODO: figure out how to handle other Client scopes...
+    # TODO: what about (soft) deletes of both users and clients
```
Comment (eanders) [inline app/jobs/materialize_permissions.rb:16]: Users and clients are soft-deleted, though clients may eventually disappear completely.

File: app/jobs/materialize_permissions.rb [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    # TODO: do we add FK to the UserClientPermission model (which will slow things down a lot)
+    # TODO: do we keep "deleted_at" records to provide audit history?
```
Comment (eanders) [inline app/jobs/materialize_permissions.rb:18]: This would be ok if we do it in such a way that we aren't completely re-writing the table nightly.

Comment (eanders) [reply app/jobs/materialize_permissions.rb:18]: Not sure how necessary it is though.  We log access to client records, which is more important.

**Pull request merged**